src/components/implementation/no_interface/llbooter/llbooter.c

```diff
@@ -299,6 +320,64 @@ comps_init(void)
 	return;
 }
 
+void 
+chkpt_comp_init(struct crt_comp *comp, struct crt_chkpt *chkpt, char *name)
+{
+#ifdef ENABLE_CHKPT
+	/* create the component */
+	void *elf_hdr;
+	int   keylen;
+	compid_t id = crt_ncomp() + 1;
+	const char *root = "binaries/";
+	int   len  = strlen(root);
+	char  path[INITARGS_MAX_PATHNAME];
+
+	assert(id < MAX_NUM_COMPS && id > 0 && name);
```

Comment:
Personally I would prefer do an assert right after the assign. For example
``` c
compid_t id;
char  path[INITARGS_MAX_PATHNAME];

id = crt_ncomp() + 1;
assert(id < MAX_NUM_COMPS && id > 0 && name);

memset(path, 0, INITARGS_MAX_PATHNAME);
strncat(path, root, len);
assert(path[len] == '\0');
```

Reply:
Yeah, this looks cleaner to me as well.